src/util/locktime.rs

```diff
@@ -0,0 +1,503 @@
+
+    fn try_from(n: u32) -> Result<Time, Self::Error> {
+        Self::new(n)
+    }
+}
+
+impl From<Time> for u32 {
+    fn from(t: Time) -> Self {
+        t.0
+    }
+}
+
+impl fmt::Display for Time {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+/// Returns true if `n` encodes a block height i.e., is < 500,000,000.
+pub fn encodes_height(n: u32) -> bool {
+    n < LOCKTIME_THRESHOLD
+}
+
+/// Returns true if `n` encodes a UNIX timestamp i.e., is >= 500,000,000.
+pub fn encodes_time(n: u32) -> bool {
+    n >= LOCKTIME_THRESHOLD
+}
+
+impl fmt::Display for LockTime {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use LockTime::*;
+
+        if f.alternate() {
+            match *self {
+                Blocks(ref h) => write!(f, "block-height {}", h),
+                Seconds(ref t) => write!(f, "block-time {} (seconds since epoch)", t),
+            }
+        } else {
+            match *self {
+                Blocks(ref h) => write!(f, "{}", h),
+                Seconds(ref t) => write!(f, "{}", t),
+            }
+        }
+    }
+}
+
+impl Encodable for LockTime {
+    fn consensus_encode<W: io::Write>(&self, e: W) -> Result<usize, io::Error> {
+        let v = self.to_raw_u32();
+        v.consensus_encode(e)
+    }
+}
+
+impl Decodable for LockTime {
+    #[inline]
+    fn consensus_decode<D: io::Read>(d: D) -> Result<Self, encode::Error> {
+        let v = u32::consensus_decode(d)?;
+        Ok(LockTime::from(v))
+    }
+}
+
+/// Catchall type for errors that relate to timelocks.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[non_exhaustive]
+pub enum Error {
+    /// An error occurred while converting a `u32` to a lock time variant.
+    Conversion(ConversionError),
+    /// An error occurred while operating on lock times.
+    Operation(OperationError),
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use self::Error::*;
+
+        match *self {
+            Conversion(e) => write_err!(f, "error converting lock time value"; e),
+            Operation(e) => write_err!(f, "error during lock time operation"; e),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+#[cfg_attr(docsrs, doc(cfg(feature = "std")))]
+impl std::error::Error for Error {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        use self::Error::*;
+
+        match *self {
+            Conversion(ref e) => Some(e),
+            Operation(ref e) => Some(e),
+        }
+    }
+}
+
+impl From<ConversionError> for Error {
+    fn from(e: ConversionError) -> Self {
+        Error::Conversion(e)
+    }
+}
+
+impl From<OperationError> for Error {
+    fn from(e: OperationError) -> Self {
+        Error::Operation(e)
+    }
+}
+
+/// An error that occurs when converting a `u32` to a lock time variant.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
`(Partial)Ord` doesn't seem to make sense here. I'm also not sure if we should commit to `Copy` just yet.

Reply:
Removed all three, thanks.